tests/benchmark_api.py

```diff
@@ -0,0 +1,655 @@
+
+    def _benchmark_set_up(self):
+        """Create a TBox and set it as the default ontology.
+
+        The new TBox contains SIMPHONY, OWL, RDFS and City.
+        """
+        self.ontology = Session(identifier="test-tbox", ontology=True)
+        self.ontology.load_parser(OntologyParser.get_parser("city"))
+        self.prev_default_ontology = Session.default_ontology
+        Session.default_ontology = self.ontology
+
+        from simphony_osp.namespaces import city
+
+        self.city_namespace = city
+
+        fr = city.City(name="Freiburg", coordinates=[0, 0])
+        citizen = city.Citizen(name="Citizen", age=25)
+        streets = tuple(
+            city.Street(name=f"street {i}", coordinates=[0, 0])
+            for i in range(self.size - 1)
+        )
+        position = random.randint(0, (self.size - 1) - 1)
+        things = list(streets)
+        things.insert(position, citizen)
+        things = tuple(things)
+        rel = {position: city.hasInhabitant}
+        for i, thing in enumerate(things):
+            fr.connect(thing, rel=rel.get(i, city.hasPart))
+
+        self.city = fr
+
+    def _benchmark_iterate(self, iteration: int = None):
+        tuple(self.city.get(oclass=self.city_namespace.Citizen))
+
+    def _benchmark_tear_down(self):
+        """Restore the previous default TBox."""
+        Session.default_ontology = self.prev_default_ontology
+
+
+def benchmark_individual_get_byoclass(benchmark):
+    """Wrapper function for the IndividualGetByOclass benchmark."""
+    return IndividualGetByOclass.iterate_pytest_benchmark(
+        benchmark, DEFAULT_SIZE
+    )
+
+
+class IndividualIterByIdentifier(Benchmark):
+    """Benchmark getting CUDS objects by uid (of type UUID)."""
+
+    def _benchmark_set_up(self):
+        """Create a TBox and set it as the default ontology.
+
+        The new TBox contains SIMPHONY, OWL, RDFS and City.
+        """
+        self.ontology = Session(identifier="test-tbox", ontology=True)
+        self.ontology.load_parser(OntologyParser.get_parser("city"))
+        self.prev_default_ontology = Session.default_ontology
+        Session.default_ontology = self.ontology
+
+        from simphony_osp.namespaces import city
+
+        self.city = city.City(name="Freiburg", coordinates=[0, 0])
+        self.citizens = tuple(
+            city.Citizen(name=f"citizen {i}", age=25) for i in range(self.size)
+        )
+        self.identifiers = tuple(
+            citizen.identifier for citizen in self.citizens
+        )
+        for citizen in self.citizens:
+            self.city.connect(citizen, rel=city.hasInhabitant)
+        self.iterator = self.city.iter(*self.identifiers)
+
+    def _benchmark_iterate(self, iteration: int = None):
+        next(self.iterator)
+
+    def _benchmark_tear_down(self):
+        """Restore the previous default TBox."""
+        Session.default_ontology = self.prev_default_ontology
+
+
+def benchmark_individual_iter_byidentifier(benchmark):
```

Comment:
At this point maybe you can just search for `_by` and update whatever is left 😅

Reply:
Resolving due to

> Minor comment, and I see it is also in other places (also get_byrel, for instance). Doesn't really need to be fixed.